src/Swashbuckle.AspNetCore.SwaggerGen/SchemaGenerator/SchemaGenerator.cs

```diff
@@ -180,6 +185,11 @@ private OpenApiSchema GenerateSchemaForParameter(
                 schema.Default = GenerateDefaultValue(dataContract, modelType, defaultValue);
             }
 
+            if (Nullable.GetUnderlyingType(modelType) is not null && !customAttributes.OfType<RequiredAttribute>().Any())
```

Comment:
`customAttributes.OfType<RequiredAttribute>().Any()` is being evaluated twice. Instead, cache the result into a local variable on the first access.

Reply:
Similarly, `Nullable.GetUnderlyingType(modelType)` is evaluated twice.

Reply:
I've realised that the second evaluation for required isn't required as modelType would always be a non nullable type at that stage.

Nullable.GetUnderlyingType(modelType) I feel is more complex
* if it's required it will be called at least once, or possibly twice with the second call using the non nullable type
* if it's not required, it could be called either zero times or once depending on whether reference is null

I won't know until after the schema is generated whether or not it will need called. I could store the result above in a nullable boolean and do `isNullable == true || Nullable.GetUnderlyingType(modelType) is not null`, but it doesn't feel like it's worth it.